README.md

```diff
@@ -4,14 +4,47 @@
 
 `$ npm install @aerogear/aerogear-reactnative-push --save`
 
-### Mostly automatic installation
+## Configuration
 
-`$ react-native link @aerogear/aerogear-reactnative-push`
+The ReactNative Unified Push Client requires configuration of the native projects to consume push messages. Please refer to your platform's documentation or the [Unified Push documentation](http://aerogear.github.io/aerogear-unifiedpush-server/) for details.
 
-## Usage
+## Usage Example
 ```javascript
-import RnUnifiedPush from '@aerogear/aerogear-reactnative-push';
+import RNUnifiedPush from '@aerogear/aerogear-reactnative-push';
 
-// TODO: What to do with the module?
-RnUnifiedPush;
+const ups = new RNUnifiedPush();
+
+ups.init(//Initialization provides the UPS address and variant secrets
+{ 
+pushServerURL: "http://10.1.10.51:9999/",
+ios: {
+    variantID: "91c039f9-d657-49cd-b507-cb78bea786e3",
+    variantSecret: "4b7fd0b4-58b5-46e8-80ef-08a6b8d449cd"
+}, 
+android: {
+    senderID: "557802659713",
+    variantID: "c046f7b6-cf86-4ae6-9eb0-bb39104ea38b",
+    variantSecret: "4907e063-cd39-477f-8938-2d817eabff97"
+}
+}).then(//Registration connects your device to UnifiedPush
```

Comment:
The application will either run on android or iOS, so the user will probably need to use [platform specific code](https://reactnative.dev/docs/platform-specific-code) to use the `android` key on android and the `ios` key on ios.
I don't think we should provide an example where both are specified at the same time.
WDYT?

Reply:
I think the answer is "it depends". We create the push-config files with multiple variants inside of them from the UPS admin ui. We also have separate steps for initialization and registration. Initialization can be used with common code pretty readily, registration does have platform divergence, but that is mostly abstracted from the user. Message handlers otoh must use platform specific code.

So here's my thought
1) Use shared code for initialization and registration
2) use shared code to set the message handler, but....
3) message handlers are defined in platform specific code and exported from their files.

```javascript handler.ios.ja
export default (message) => {ios}
```

```javascript handler.android.ja
export default (message) => {android}
```

```javascript app.js

import handler from './handler.js';
import ups from 'blach'

ups.init({}).then(ups.register).then(()=>ups.messageHandler(handler));
```

(Obvious pseduocode is obvious pseudocode, obviously)

wdyt about that approach?

Reply:
👍  Agree!

Reply:
https://github.com/aerogear/unifiedpush-cookbook/pull/7

This has an implementation of the rector, wdyt?